sponnet/pipeline.libsonnet

```diff
@@ -1,12 +1,15 @@
 {
   pipeline():: {
-    limitConcurrent: true,
     keepWaitingPipelines: false,
+    limitConcurrent: true,
     notifications: [],
     stages: [],
     triggers: [],
     withApplication(application):: self + { application: application },
     withExpectedArtifacts(expectedArtifacts):: self + if std.type(expectedArtifacts) == 'array' then { expectedArtifacts: expectedArtifacts } else { expectedArtifacts: [expectedArtifacts] },
+    withId(id):: self + { id: id },
```

Comment:
Seems like this goes way back to https://github.com/spinnaker/spin/pull/5, but wasn't really well explained. I think we should honor the id if it's there but not require it. I think orca/front50 will just generate a uuid if it's missing in `spin`. As far as sponnet goes we should have an option for an `id` but not make it a requirement.

Reply:
Okay. I can remove setting the pipeline id in the demo and update this PR: https://github.com/spinnaker/spinnaker/pull/3635/files#diff-01c380160d0d012b77fc2bdbbded0996R165

Or, we can leave it in and I can remove in a later PR once other changes are merged?
I expect that I'll be testing out the new Kubernetes stages when 1.11 and will end up adding those to sponnet. Whatever is not too much hassle for reviewers. 
Please sing out if batching too much/little in any case.

Reply:
I think we should leave this in for now since the released spin works that way and change it once the `spin` PRs relaxing the id constraint is merged.

Reply:
Sounds good.